src/event_loop.rs

```diff
@@ -378,6 +262,72 @@ impl<T> EventLoopWindowTarget<T> {
         #[cfg(any(x11_platform, wasm_platform, wayland_platform, windows))]
         self.p.listen_device_events(_allowed);
     }
+
+    /// When the current loop iteration finishes, immediately begin a new iteration regardless of
+    /// whether or not new events are available to process.
+    ///
+    /// This is the default.
+    pub fn set_poll(&self) {
+        self.p.set_poll()
+    }
+
+    /// When the current loop iteration finishes, suspend the thread until another event arrives.
+    pub fn set_wait(&self) {
+        self.p.set_wait()
+    }
+
+    /// When the current loop iteration finishes, suspend the thread until either another event
+    /// arrives or the given time is reached.
+    ///
+    /// Useful for implementing efficient timers. Applications which want to render at the display's
+    /// native refresh rate should instead use [`set_poll()`] and the VSync functionality of a graphics API
+    /// to reduce odds of missed frames.
+    ///
+    /// [`set_poll()`]: Self::set_poll
+    pub fn set_wait_until(&self, instant: Instant) {
+        self.p.set_wait_until(instant)
+    }
+
+    /// Sets this to wait until a timeout has expired.
+    ///
+    /// In most cases, this is uses [`set_wait_until()`]. However, if the timeout overflows, it
+    /// instead uses [`set_wait()`].
+    ///
+    /// [`set_wait_until()`]: Self::set_wait_until
+    /// [`set_wait()`]: Self::set_wait
+    pub fn set_wait_timeout(&self, timeout: Duration) {
+        match Instant::now().checked_add(timeout) {
+            Some(instant) => self.set_wait_until(instant),
+            None => self.set_wait(),
+        }
+    }
+
+    /// Send a [`LoopExiting`] event and stop the event loop. This variant is *sticky* - once set,
+    /// it cannot be changed, and any future attempts to do so will
+    /// result in it being reset to it's previous state.
+    ///
+    /// The contained number will be used as exit code. The [`set_exit()`] method is a shortcut for this
+    /// with exit code 0.
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **Android / iOS / Web:** The supplied exit code is unused.
+    /// - **Unix:** On most Unix-like platforms, only the 8 least significant bits will be used,
+    ///   which can cause surprises with negative exit values (`-42` would end up as `214`). See
+    ///   [`std::process::exit`].
+    ///
+    /// [`LoopExiting`]: Event::LoopExiting
+    /// [`set_exit()`]: Self::set_exit
+    pub fn set_exit_with_code(&self, code: i32) {
+        self.p.set_exit_with_code(code)
+    }
+
+    /// Alias for [`set_exit_with_code()`]`(0)`.
+    ///
+    /// [`set_exit_with_code()`]: Self::set_exit_with_code
+    pub fn set_exit(&self) {
```

Comment:
I think this one makes sense to call `exit` instead of `set_exit`; even though it internally works like the others, it's much more understandable as an "action" rather than setting some state.

Reply:
I'm gonna leave this to later until we have a consensus if we want getters or not.

Reply:
Did exit codes just vanish? Not that I need them.

Reply:
Yes.
See https://github.com/rust-windowing/winit/pull/3056#discussion_r1306728448.

Reply:
I guess we will figure out getters in #3064.
Currently my hope is to be able to add back the enum and just expose a setter and getter for the enum.

In the meantime I have renamed `set_exit()` to `exit()`.

Reply:
You can't really have the `getters` and `setters` on the enum, they are part of the `EventLoop`. The getter on the `EventLoopWindowTarget` should just return a `ControlFlow`, we could also unify the way `ControlFlow` is being set, so it's just a method taking a `ControlFlow` in.

We can't have a separate methods on the `enum`, simply because we need to maintain state for the `exit` thing, but as it was said, `Exit` is a separate concept, and likely doesn't need to be merged in the `ControlFlow` exposed to the user, and event internally it should likely be just a bool `exiting` stored on the `EventLoopWindowTarget` or `EventLoop` directly.

Reply:
> You can't really have the `getters` and `setters` on the enum, they are part of the `EventLoop`. The getter on the `EventLoopWindowTarget` should just return a `ControlFlow`, we could also unify the way `ControlFlow` is being set, so it's just a method taking a `ControlFlow` in.

That is what I meant!